src/containers/LoginRegister/index.jsx

```diff
@@ -0,0 +1,224 @@
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      await auth.signInWithEmailAndPassword(email, password);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  const passwordReset = async (email) => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      await auth.sendPasswordResetEmail(email);
+      setLoading(false);
+      setMessage(
+        <Alert
+          style={{ marginBottom: 10 }}
+          showIcon
+          message={
+            <div>
+              {t("loginRegister.anEmailsent")}
+              <b>{email}</b>
+              {t("loginRegister.anEmailsentcontain")}
+            </div>
+          }
+          type="success"
+        ></Alert>
+      );
+      setActive("login");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  // Login with FACEBOOK
+  const loginWithFacebook = async () => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      let provider = new firebase.auth.FacebookAuthProvider();
+      provider.setCustomParameters({
+        display: "popup",
+      });
+      let u = await auth.signInWithPopup(provider);
+      console.log("facebook user: ", u.user);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  // Loogin with GOOGLE
+  const loginWithGoogle = async () => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      let provider = new firebase.auth.GoogleAuthProvider();
+      provider.addScope("https://www.googleapis.com/auth/contacts.readonly");
+      let u = await auth.signInWithPopup(provider);
+      console.log("google user: ", u.user);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+
+  const register = async (fullname, email, age, password) => {
+    console.log(email, password);
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      let registeredUser = await auth.createUserWithEmailAndPassword(
+        email,
+        password
+      );
+      await firestore.collection("users").doc(registeredUser.user.uid).set({
+        uid: registeredUser.user.uid,
+        fullname: fullname,
+        age: age,
+        userExperiencePoints: 0,
+      });
+      await auth.signInWithEmailAndPassword(email, password);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+      console.log(e.message);
+    }
+  };
+  const [active, setActive] = useState("login");
+  const [error, setError] = useState("");
+  const [message, setMessage] = useState();
+  const [loading, setLoading] = useState(false);
+  const { user } = useContext(AuthContext);
+  console.log("User ", user);
```

Comment:
fix, I'm going to stop reviewing now, proofread the PR and get 2 reviews then send again